src/examples/zkprogram/program-with-non-pure-input.ts

```diff
@@ -0,0 +1,34 @@
+import { Field, Provable, Struct, ZkProgram, assert } from 'o1js';
+
+class MyStruct extends Struct({
+  label: String,
+  value: Field,
+}) {}
+
+let MyProgram = ZkProgram({
+  name: 'example-with-non-pure-inputs',
+  publicInput: MyStruct,
+  publicOutput: MyStruct,
+
+  methods: {
+    baseCase: {
+      privateInputs: [],
+      async method(input: MyStruct) {
+        return {
```

Comment:
This works! But right now we don't handle the case where the aux values are changed _in_ the circuit, like this

```ts
input.label = "in circuit";
return {
   publicOutput: input
}

// after the circuit
assert(input.label == "in circuit")
```


so either we make this work or we explcitily tell developers not to touch the auxiliary values in the circuit; I prefer the first approach

Reply:
We need them in the circuit!